content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/ContentEntityProvider.java

```diff
@@ -179,6 +181,10 @@ public boolean entityExists(String id) {
 						ContentItem item = new ContentItem();
 						item.setType("collection");
 						item.setSize(contentHostingService.getCollectionSize(id));
+						if (allowUpdateSite) // to be consistent with UI
+							item.setQuota(Long.toString(contentHostingService.getQuota(collection)));
+						item.setUsage(Long.toString(collection.getBodySizeK() * 1024));
```

Comment:
Yeah, the columns in the database do exist, but the call to getBodySizeK has to run a select/aggregate function (SUM) on all of the contents in that collection to get the size, and I didn't see anywhere that this sum is cached.

select SUM(FILE_SIZE) from CONTENT_RESOURCE where CONTEXT = ?

For the regular case, probably not too bad to run that, but I've seen huge content collections. ;)